### ISSUE TITLE
Check if non-interactive ssh works for host and notify user if not

### ISSUE THREAD

**Issue by viccuad** [openSUSE/repose#52]
Hi,

A colleage has notified me that all `repose` functionality doesn't work
if you use it with an ssh setup that spawns a passphrase  prompt. In fact, it is documented
on its [manpage](https://github.com/openSUSE/repose/blob/master/man/repose.1.in#L121) like that:

```
repose uses ssh(1) to access remote machines.  "ssh user@hostname" must work without prompting for any passwords or passphrases.
```

This issue has made me file other issues in the past (#49, #48 , #3).

It would be nice to have `repose` work, (or at least notify in some, any way why it fails) if
the passphrase is requested. 

If that's not possible, it would be nice to have that silent fail documented in the manpage,
and also a possible solution, for example catching the passphrase with ssh-agent, or using
ssh-ident and the following alias:

```
alias repose='BINARY_SSH=repose $ZSH/suse/nostow/bin/ssh-ident'
```

Cheers.

**Comment by mimi1vx** [comment]
It uses ssh in batch mode --> so public key auth .. and this is also in repose docu. 

>  repose uses ssh(1) to access remote machines.  "ssh user@hostname" must work without prompting for any passwords or passphrases.

is too hard to use `ssh-copy-id` after installation REFHOST ?

**Comment by viccuad** [comment]
> It uses ssh in batch mode --> so public key auth .. and this is also in repose docu.
>>    repose uses ssh(1) to access remote machines. "ssh user@hostname" must work without prompting for any passwords or passphrases.

Please reread my post, I have already said that. In fact, I have even already quoted exactly
the same part of the manpage that you have quoted in your message. At this point you are
quoting me.

> is too hard to use ssh-copy-id after installation REFHOST ?

My ssh id was already present in the system and I could ssh prior to using `repose`.
Also, using ssh-copy-id is totally irrelevant with having `repose` not work if an ssh key
passphrase is asked for.
Have you maybe confused "ssh key passhphrase" with "root login password" in your
reply?

I kindly ask why this feature request is marked as invalid and has been
closed, I don't see any reason for that.

**Comment by plusky** [comment]
Well issue of silent fails is already adressed/discussed at #2, as for documenting this limitation - documentation, quoted here several times already clearly states `without prompting for any passwords or passphrases.` which covers need for decrypting ssh keys on side of user.

This behavior will of course make usage difficult for people who are using encrypted ssh keys and at same time are not using ssh-agent or some of it's alternatives.

**Comment by drpaneas** [comment]
> issue of silent fails is already adressed/discussed at #2

Unfortunatelly, #2 has been marked as **wontfix**, which means that although it might be technically possible, it doesn't worth spending development time on it at the moment. In other words, the message I get is: _feel free to investigate the problem yourself and submit a patch if you succeed._ -- which is acceptable..

However, the case here is _different_, as this issue has been marked as **invalid** and not as **wontfix**. 

> as for documentation [...] covers the need for decrypting ssh keys on side of user.

That's one way of interpretation. The other way is that the tools is so insufficiently designed that we have to document its flaws instead of fixing them. As a result of this _certain_ logic, we end up labelling valid bugs as invalid. I would like to point out that although I see why you did it, I am afraid I don't agree that invalidating a bug in expense of common sense, is the right approach for the development of _repose_. 

Simply put, _no matter_ if this _is-or-not_ documented at all, the tool should have warned the user (in this case, _Victor_) about its inability to connect to the ref-hosts. That's what the average Joe would have expected; that's basic Computer Science 101.  

> This issue has made me file other issues in the past (#49, #48 , #3).

And this is just the beginning... As long as _repose_ doesn't provide the very basic and expected user feedback, we will get more false bug reports from new users, either colleagues or community members. Be prepared for it :)

**Comment by plusky** [comment]
Indeed we might want to check for ability to ssh non-interactively before starting non-interactive operations - lets explore this possibility.
